File: Buddy/buddy-flutter/lib/models/pet_action.dart
enum PetAction {
  idle,
  eat,
  sleep,
  play,
  sad;

  String get raw => name;

  String get label => switch (this) {
        PetAction.idle  => 'tranquilo',
        PetAction.eat   => 'comiendo',
        PetAction.sleep => 'durmiendo',
        PetAction.play  => 'jugando',
        PetAction.sad   => 'triste',
      };

  String get emoji => switch (this) {
        PetAction.idle  => '✨',
        PetAction.eat   => '🍖',
        PetAction.sleep => '💤',
        PetAction.play  => '🎾',
        PetAction.sad   => '😢',
      };

  /// 4×4 sheet layout: row 0 idle | row 1 walk | row 2 eat | row 3 sleep
  ({int row, List<int> frames}) get spriteFrames => switch (this) {
        PetAction.idle  => (row: 0, frames: const [0, 1, 2, 3]),
        PetAction.eat   => (row: 2, frames: const [0, 1, 2, 3]),
        PetAction.sleep => (row: 3, frames: const [0, 1, 2, 3]),
        PetAction.play  => (row: 1, frames: const [0, 1, 2, 3]),
        PetAction.sad   => (row: 0, frames: const [0, 1]),
      };

  static PetAction fromRaw(String? r) =>
      PetAction.values.firstWhere((e) => e.name == r, orElse: () => PetAction.idle);
}

File: Buddy/buddy-ios/Sources/buddy/Models/PetAction.swift
import Foundation

enum PetAction: String, Codable, CaseIterable {
    case idle
    case eat
    case sleep
    case play
    case sad

    var label: String {
        switch self {
        case .idle:  "tranquilo"
        case .eat:   "comiendo"
        case .sleep: "durmiendo"
        case .play:  "jugando"
        case .sad:   "triste"
        }
    }

    /// 4×4 sheet layout: row 0 idle | row 1 walk | row 2 eat | row 3 sleep
    /// Other actions (play, sad) reuse closest visual fit.
    var spriteFrames: (row: Int, frames: [Int]) {
        switch self {
        case .idle:  (0, [0, 1, 2, 3])
        case .eat:   (2, [0, 1, 2, 3])
        case .sleep: (3, [0, 1, 2, 3])
        case .play:  (1, [0, 1, 2, 3])  // walk frames look energetic
        case .sad:   (0, [0, 1])         // idle slow
        }
    }
}
